specification/extendedlocation/resource-manager/Microsoft.ExtendedLocation/stable/2021-08-15/customlocations.json

```diff
@@ -0,0 +1,728 @@
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "minLength": 1,
+      "maxLength": 63,
+      "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]{0,61}[a-zA-Z0-9]$",
+      "description": "Custom Locations name."
+    }
+  },
+  "definitions": {
+    "EnabledResourceType": {
+      "description": "EnabledResourceType definition.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EnabledResourceTypeProperties",
+          "description": "The set of properties for EnabledResourceType specific to a Custom Location",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "Metadata pertaining to creation and last modification of the resource",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "EnabledResourceTypeProperties": {
+      "description": "Properties for EnabledResourceType of a custom location.",
+      "type": "object",
+      "properties": {
+        "clusterExtensionId": {
+          "description": "Cluster Extension ID",
+          "type": "string"
+        },
+        "extensionType": {
+          "description": "Cluster Extension Type",
+          "type": "string"
+        },
+        "typesMetadata": {
+          "description": "Metadata of the Resource Type",
+          "type": "array",
+          "items": {
+            "description": "Metadata of the Resource Type.",
+            "type": "object",
+            "properties": {
+              "apiVersion": {
+                "description": "Api Version of Resource Type",
+                "type": "string"
+              },
+              "resourceProviderNamespace": {
+                "description": "Resource Provider Namespace of Resource Type",
+                "type": "string"
+              },
+              "resourceType": {
+                "description": "Resource Type",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "EnabledResourceTypesListResult": {
+      "description": "List of EnabledResourceTypes definition.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The URL to use for getting the next set of results.",
+          "type": "string",
+          "readOnly": true
+        },
+        "value": {
+          "description": "The list of EnabledResourceTypes available for a customLocation.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/EnabledResourceType"
+          }
+        }
+      }
+    },
+    "Identity": {
```

Comment:
>Identity [](http://example.com/codeflow?start=5&length=8)

If you're planning to add user assigned identities in the future, you should consider referencing the equivalent definition from common types v2. It would allow you to add support for UAIs without breaking changes.

Reply:
We would want to reference common types but so far the common types (v2) only supports `SystemAssigned` identity, while in this PR we are adding `SystemAssigned` and `None` types. 

Common Types (v2) : https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v2/types.json#L245